### DISCUSSION TITLE
Implement some useful braille features

### DISCUSSION THREAD

**Discussion by Adriani90** [nvaccess/nvda#16879] [Ideas]
Dear all,

I have found a very interesting Repo which gives a good overview over all braille features implemented in jaws. i don't know if it is possible to write something out of it to work with NVDA, but I think it is very interesting to give it a try.
https://github.com/gloob/gpii-settings/blob/master/AT-Tools/JAWS/Default.jcf.csv

By the way, gloob has some very interesting repos related to accessibility.

In my view, very interesting features are following:
Braille.PositionOfStatusCells
Focus.PositionOfStatusCells
Braille.UseHowManyStatusCells
Focus.UseHowManyStatusCells
Braille.GeneralizeBullets
Focus.GeneralizeBullets
Braille.BrailleMode
focus.BrailleMode
Braille.BrailleSlipMode
focus.BrailleSleepMode

Braille.ReversePanningButtons
Braille.PreferredTranslationTables

.BrailleMode could contribute to solve issues when working with braille and speech simultanuously, covered in issue
#7887 

.PreferredTranslationTables could contribute to generate a script for automatic braille output by picking out preferred tables, covered in issue
#7848 

Comments are appreciated.

Thanks

**Comment by josephsl** [comment]
Hi, I’d say let Braille Extender add-on take care of this in the meantime, and then we should consider cherry-picking priority features from that add-on when the opportunity presents itself. Thanks.

**Comment by LeonarddeR** [comment]
I believe that this issue is a bit too broad in scope.

[**@Adriani90**](https://github.com/Adriani90) commented on [18 Feb 2018, 22:01 CET](https://github.com/nvaccess/nvda/issues/8011#issue-298117796 "2018-02-18T21:01:33Z - Replied by Github Reply Comments"):
> Braille.PositionOfStatusCells
> Focus.PositionOfStatusCells
> Braille.UseHowManyStatusCells
> Focus.UseHowManyStatusCells

NVDA doesn't use the concept of status cells.

> Braille.GeneralizeBullets
> Focus.GeneralizeBullets

Could you elaborate on what this does?

> Braille.BrailleMode
> focus.BrailleMode

I assume this is line mode, structured mode, etc.? We already have tethering to focus/review and several review modes.

> Braille.BrailleSlipMode
> focus.BrailleSleepMode

Why would one want to enable sleep mode for braille? I see perfectly valid reasons to have speech sleep, but haven't yet found a reason for braille to do so.

> Braille.ReversePanningButtons

This is already possible by reassigning gestures.

> Braille.PreferredTranslationTables

See also #290.

**Comment by Adriani90** [comment]
> Could you elaborate on what this does?
If there are many bullet shapes in a document, only the bullets which can be displayed in braille are shown on the display. All other bullet shapes are being generalized to for example * (dots 35) or - (dots 36). I think this issue is not very world changing but it would show in braille at least that there is a bullet.

> I assume this is line mode, structured mode, etc.? We already have tethering to focus/review and several review modes.
As far as I could experience in earlier jaws versions, there is also a speech mode which shows also the element status and type on the braille display. In surface mode element status ans type is not shown. I thought it would make sense for users who are working with braille and speech simultanuously.

> Why would one want to enable sleep mode for braille? I see perfectly valid reasons to have speech sleep, but haven't yet found a reason for braille to do so.
For example you want to freeze braille on the last application while working in another application with speech.

**Comment by josephsl** [comment]
Hi,

2024 update: my stance on collaborating with Braille Extender folks and getting needed things into NVDA Core still stands.

Thanks,

**Comment by SaschaCowley** [comment]
This has been converted to a discussion as it is too broad for an issue. Please open individual issues for the features you would like to see in NVDA.

**Comment by Adriani90** [comment]
I think @LeonarddeR has already filtered out in his comment above some of the issues,
so in my view foolowing features are still standing:
* Generalize bullets: If there are many bullet shapes in a document, only the bullets which can be displayed in braille are shown on the display. Oother bullets are ignored and the braille user doesn't know there are bullets. So All other non displayable bullet shapes could be generalized to for example (dots 35) or - (dots 36). I think this request is not very revilutionary but it would show in braille at least that there is a bullet at all.
* Braille sleep mode: I think this is rather called Braille freeze mode in Jaws. For example you want to freeze braille to show content in one application only, while working in another application with speech. I guess this could work when two windows of two different applications are shown on the screen. E.g. reading with braille a document in MS Word while using the keyboard to edit a table in MS Excel via speech.

For
Braille.ReversePanningButtons
and 
Braille.PreferredTranslationTables
@aaclause might be able to contribute with his code, bringing these two features to the core.

cc: @Emil-18, @bramd

**Comment by Emil-18** [comment]
@Adriani90 One feature I think should be implemented in core is the Undefined character representation feature from braille extender. With it, combined with the speech mode feature, there is practicly no information that is inaccessible to a braille only user. Plus, it would most likely solv Generalize bullits to a degree. As for the braille freeze feature, this is possible by tethering braille to review, set the review cursor where you want, e.g in a word document, turn off moovement of the review cursor do to focus changes, switch to e.g an excel document, and navigate that with focus while reading the word document with braille.
